mesonbuild/interpreter/kwargs.py

```diff
@@ -203,7 +203,7 @@ class Project(TypedDict):
 
     version: T.Optional[FileOrString]
     meson_version: T.Optional[str]
-    default_options: T.List[str]
+    default_options: T.Dict[OptionKey, T.Union[str, int, bool, list]]
```

Comment:
I think you should use `T.List[T.Any]`. But maybe @dcbaker has opinions? I am just looking at the definition of `TYPE_elementary`.

Reply:
In theory that should be `T.List[str]` because we don't support other types in lists. But it is a nightmare to support ContainerTypeInfo inside a ContainerTypeInfo in KwargInfo. It's easier to allow any type inside the list because UserOption class already does the validation anyway.

I would expect `T.List[T.Any]` and `list` to be synonym. TBH as long as mypy pass it's good enough for me :P

Reply:
I think we could use a validator for this? something like:
```python
def _validator(val: str | list[str] | dict[str, Any]) -> str | None:
    if not isinstance(val, dict):
        return None
    for k, v in val.items():
        if not isinstance(v, str):
            return f'invalid value for key {k}, which was of type {type(v)}, but should have been a string'
    return None
```